src/applications/bmqtool/m_bmqtool_inpututil.cpp

```diff
@@ -319,5 +501,57 @@ bool InputUtil::populateSubscriptions(bmqt::QueueOptions*              out,
     return !failed;
 }
 
+bool InputUtil::decodeHexDump(bsl::ostream& out,
+                              bsl::istream& in,
+                              bsl::string*  error)
+{
+    char outputBuffer[4];  // should be equal to k_BLOCK_SIZE in
+                           // bdlb::Print::hexDump()
+    bdlde::HexDecoder hexDecoder;
+
+    bsl::string line;
+    int         numOut = 0;
+    int         numIn  = 0;
+
+    while (bsl::getline(in, line)) {
+        if (line.empty())
+            continue;  // skip empty lines
```

Comment:
This is discussable, but consider adding brackets `{}` here and in other places.
Motivation: avoid cases like this

```
if (condition)
    smth1();
    smth2();  // smth2() is executed unconditionally despite formatting
```

```
if (condition)
//    smth1();  // commented out, and unexpectedly smth2() now is under condition despite formatting
smth2();
```